doc/dash/dash-sonic-hld.md

```diff
@@ -184,6 +185,7 @@ DASH Sonic implementation is targeted for appliance scenarios and must handles m
 14. During a bulk operation, if any part/subset of API fails, implementation shall return *error* for the entire API. Sonic implementation shall validate the entire API as pre-checks before applying and return accordingly.
 15. Implementation must have flexible memory allocation for ENI and not reserve max scale during initial create (e.g 100k routes). This is to allow oversubscription.
 16. Implementation must not have silent failures for APIs. E.g accepting an API from controller, returning success and failing in the backend. This is orthogonal to the idempotency of APIs described above for ADD and Delete operations. Intent is to ensure SDN controller and Sonic implementation is in-sync
+17. If previously bound route group is unbound from all ENIs, it will be implicitly deleted. Since memory constraints mean that routes cannot be cached indefinitely in orchagent/SAI, route information is lost after programming to hardware. This means it is not possible to recreate the routes for a given route group, so once a route group is completely unbound it cannot be rebound to another ENI and must be deleted. Adding routes for a route group which has been deleted in this manner will result in an error.
```

Comment:
@mukeshmv , could you review/signoff?

Reply:
Correct, and will also remove it from orchagent internal state/cache